rig-neo4j/src/lib.rs

```diff
@@ -162,19 +176,165 @@ impl Neo4jClient {
     /// See the Neo4j [documentation (Create vector index)](https://neo4j.com/docs/genai/tutorials/embeddings-vector-indexes/setup/vector-index/) for more information on creating indexes.
     ///
     /// ❗IMPORTANT: The index must be created with the same embedding model that will be used to query the index.
-    pub fn index<M: EmbeddingModel>(
+    pub async fn get_index<M: EmbeddingModel>(
         &self,
         model: M,
-        index_config: IndexConfig,
+        index_name: &str,
         search_params: SearchParams,
-    ) -> Neo4jVectorIndex<M> {
-        Neo4jVectorIndex::new(self.graph.clone(), model, index_config, search_params)
+    ) -> Result<Neo4jVectorIndex<M>, VectorStoreError> {
+        #[derive(Deserialize)]
+        struct IndexInfo {
+            name: String,
+            properties: Vec<String>,
+            options: IndexOptions,
+        }
+
+        #[derive(Deserialize)]
+        #[serde(rename_all = "camelCase")]
+        struct IndexOptions {
+            _index_provider: String,
+            index_config: IndexConfigDetails,
+        }
+
+        #[derive(Deserialize)]
+        struct IndexConfigDetails {
+            #[serde(rename = "vector.dimensions")]
+            vector_dimensions: i64,
+            #[serde(rename = "vector.similarity_function")]
+            vector_similarity_function: String,
+        }
+
+        let index_info = Self::execute_and_collect::<IndexInfo>(
+            &self.graph,
+            neo4rs::query(Self::GET_INDEX_QUERY).param("index_name", index_name),
+        )
+        .await?;
+
+        let index_config = if let Some(index) = index_info.first() {
+            if index.options.index_config.vector_dimensions != model.ndims() as i64 {
+                tracing::warn!(
+                    "The embedding vector dimensions of the existing Neo4j DB index ({}) do not match the provided model dimensions ({}). This may affect search performance.",
+                    index.options.index_config.vector_dimensions,
+                    model.ndims()
+                );
+            }
+            IndexConfig::new(index.name.clone())
+                .embedding_property(index.properties.first().unwrap())
+                .similarity_function(VectorSimilarityFunction::from_str(
+                    &index.options.index_config.vector_similarity_function,
+                )?)
+        } else {
+            let indexes = Self::execute_and_collect::<String>(
+                &self.graph,
+                neo4rs::query(Self::SHOW_INDEXES_QUERY),
+            )
+            .await?;
+            return Err(VectorStoreError::DatastoreError(Box::new(
+                std::io::Error::new(
+                    std::io::ErrorKind::NotFound,
+                    format!(
+                        "Index `{}` not found in database. Available indexes: {:?}",
+                        index_name, indexes
+                    ),
+                ),
```

Comment:
This error construction seems complicated, it seems cleaner to just pass a `format!(...).into()` directly to the `DatastoreError`.